### PR TITLE
CB-8156 Call error callback on pickContact cancellation

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
Also introduce OPERATION_CANCELLED_ERROR as an error code for reporting contact picker cancellation.

See [CB-8156](https://issues.apache.org/jira/browse/CB-8156) for details

### CODE REVIEW

File: src/android/ContactManager.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ Licensed to the Apache Software Foundation (ASF) under one
     public static final int PENDING_OPERATION_ERROR = 3;
     public static final int IO_ERROR = 4;
     public static final int NOT_SUPPORTED_ERROR = 5;
+    public static final int OPERATION_CANCELLED_ERROR = 5;
```
Comment (purplecabbage) [inline src/android/ContactManager.java]: Isn't this supposed to be 6 as above, so below?

File: src/windows/ContactProxy.js [hunk 2]
```diff
@@ -242,7 +242,9 @@ module.exports = {
         pickRequest.done(function (contact) {
             // if contact was not picked
             if (!contact) {
-                fail && fail(new Error("User did not pick a contact."));
```
Comment (purplecabbage) [inline src/windows/ContactProxy.js]: This looks good.  The implied check for `fail != null` here was unnecessary as ALL calls to exec proxies have valid success/error function passed in.

**Comment by purplecabbage** [comment]
Looks good, except for the java implementation (re)used error code 5 instead of the 6 used everywhere else. The added tests should reveal this.

**Comment by purplecabbage** [comment]
One more thing ...
Please add a test for the new constant to 
`contacts.spec.25 ContactError constants should be defined` at /tests/tests.js#L473

**Comment by vladimir-kotikov** [comment]
Updated. Thanks for reviewing, @purplecabbage

**Pull request merged**